RecoTracker/PixelSeeding/plugins/alpaka/CAHitNtuplet.cc

```diff
@@ -358,14 +359,44 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
     auto const& geometry = runCache()->get(iEvent.queue());
     auto const& hits = iEvent.get(tokenHit_);
 
-    std::array<double, 1> nHitsV = {{double(hits.nHits())}};
-    std::array<double, 1> emptyV;
-
-    uint32_t const maxTuples = maxNumberOfTuples_.evaluate(nHitsV, emptyV);
-    uint32_t const maxDoublets = maxNumberOfDoublets_.evaluate(nHitsV, emptyV);
+    /// Don't bother if no hits on BPix1 and no good graph for that
+    /// (so no staring pair without BPix1 as first layer).
+    /// TODO: this could be extended to a more general check for
+    /// no hits on any of the starting layers.
+
+    bool hitsOnBPix1 = hits.offsetBPIX2() > 0;
+    bool startNoBPix1 = hitsOnBPix1;  // one would suffice, two for readability
+    if (not hitsOnBPix1) {
+      startNoBPix1 = false;
+      for (const unsigned int& i : globalCache()->startingPairs_) {
+        if (globalCache()->pairGraph_[2 * i] > 0) {
+          startNoBPix1 = true;
+          break;
+        }
+      }
+    }
 
-    iEvent.emplace(tokenTrack_,
-                   deviceAlgo_.makeTuplesAsync(hits, geometry, bf, maxDoublets, maxTuples, iEvent.queue()));
+    if (startNoBPix1) {
+      std::array<double, 1> nHitsV = {{double(hits.nHits())}};
+      std::array<double, 1> emptyV;
+
+      uint32_t const maxTuples = maxNumberOfTuples_.evaluate(nHitsV, emptyV);
+      uint32_t const maxDoublets = maxNumberOfDoublets_.evaluate(nHitsV, emptyV);
+
+      iEvent.emplace(tokenTrack_,
+                     deviceAlgo_.makeTuplesAsync(hits, geometry, bf, maxDoublets, maxTuples, iEvent.queue()));
+
+    } else {
+      edm::LogWarning("CAHitNtupletAlpaka") << "No hit on BPix1 (" << hits.offsetBPIX2()
+                                            << ") and all the starting pairs has BPix1 as inner layer.\nIt's useless "
+                                               "to run the CA. Returning with 0 tracks!";
+      auto& queue = iEvent.queue();
+      const auto device = alpaka::getDev(queue);
+      reco::TracksSoACollection tracks({{0, 0}}, queue);
+      auto ntracks_d = cms::alpakatools::make_device_view(device, tracks.view().nTracks());
```

Comment:
you should be able to pass directly the `queue`:
```suggestion
      auto ntracks_d = cms::alpakatools::make_device_view(queue, tracks.view().nTracks());
```